x/rewards/mintbankkeeper/keeper.go

```diff
@@ -0,0 +1,77 @@
+package mintbankkeeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authTypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	mintTypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+
+	"github.com/archway-network/archway/pkg"
+	rewardsTypes "github.com/archway-network/archway/x/rewards/types"
+)
+
+var _ mintTypes.BankKeeper = Keeper{}
+
+// RewardsKeeperExpected defines the expected interface for the x/rewards keeper.
+type RewardsKeeperExpected interface {
+	InflationRewardsRatio(ctx sdk.Context) sdk.Dec
+	TrackInflationRewards(ctx sdk.Context, rewards sdk.Coin)
+}
+
+// Keeper is the x/bank keeper decorator that is used by the x/mint module.
+// Decorator is used to split inflation tokens between the rewards collector and the fee collector accounts.
+type Keeper struct {
+	bankKeeper    mintTypes.BankKeeper
+	rewardsKeeper RewardsKeeperExpected
+}
+
+// NewKeeper creates a new Keeper instance.
+func NewKeeper(bk mintTypes.BankKeeper, rk RewardsKeeperExpected) Keeper {
+	return Keeper{
+		bankKeeper:    bk,
+		rewardsKeeper: rk,
+	}
+}
+
+// SendCoinsFromModuleToModule implements the mintTypes.BankKeeper interface.
+func (k Keeper) SendCoinsFromModuleToModule(ctx sdk.Context, senderModule, recipientModule string, amt sdk.Coins) error {
+	// Perform the split only if the recipient is fee collector (which for instance is always the case) and
+	// inflation rewards are enabled.
+	ratio := k.rewardsKeeper.InflationRewardsRatio(ctx)
+	if recipientModule != authTypes.FeeCollectorName || ratio.IsZero() {
+		return k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, recipientModule, amt)
+	}
+
+	dappRewards, stakingRewards := pkg.SplitCoins(amt, ratio)
+
+	// Send to the x/auth fee collector account
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, recipientModule, stakingRewards); err != nil {
+		return err
+	}
+
+	// Send to the x/rewards account
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, senderModule, rewardsTypes.ContractRewardCollector, dappRewards); err != nil {
+		return err
+	}
+
+	// Check that only one coin has been minted
+	if len(dappRewards) != 1 {
```

Comment:
Why are we making an assumption here that only one coin can be minted? 

I understand current version of cosmos sdk only support one coin for inflation, but that could change in future. Besides, if we go by that then input will always contain one coin anyway. So there is no point in this check.

Reply:
@fdymylja Thoughts?

Reply:
Copied from your code so need yours feedback.

Reply:
This mocks the behaviour of the current minting module in the SDK, once we upgrade to a different SDK version if there are changes in the minting module we can change this.